tests/test_date.py

```diff
@@ -2,66 +2,136 @@
 
-        for begin, end, expected in fixtures:
-            result = list(date.date_range(begin, end, months=1))
-            self.assertEquals(expected,
-                              [(d.year, d.month) for d in result])
+    def given_period_size(self, **params):
+        self.period_size = params
 
-    def test_should_reject_easily_mistaken_dateutil_arguments(self):
-        begin = datetime(2014, 6, 15)
-        end = datetime(2014, 6, 25)
+    def when_month_range_generated(self):
+        self.generated_months = list(date.date_range(self.period_start, self.period_end, **self.period_size))
+
+    def when_date_range_generated(self):
+        self.range = list(date.date_range(self.period_start, self.period_end))
+
+    def when_period_is_parsed(self):
+        try:
+            self.result = date.date_range(self.period_start, self.period_end, **self.period_size).next()
+        except Exception as error:
+            self.result = error
+
+    def then_expected_months_are(self, expected):
+        self.assertEqual(expected,
+                         [(d.year, d.month) for d in self.generated_months])
+
+    def then_range_length_is(self, expected_length):
+        self.assertEqual(expected_length, len(self.range))
 
-        with self.assertRaisesRegexp(ValueError, "Invalid argument"):
-            date.date_range(begin, end, month=1).next()
+    def then_all_dates_are_present_in_range(self):
+        date_under_test = self.period_start
+        while date_under_test < self.period_end:
+            self.assertIn(date_under_test, self.range)
+            date_under_test += timedelta(days=1)
 
-        with self.assertRaisesRegexp(ValueError, "Invalid argument"):
-            date.date_range(begin, end, day=1).next()
+    def then_range_is_in_ascending_order(self):
+        for i in xrange(len(self.range) - 1):
+            self.assertLess(self.range[i], self.range[i + 1])
+
+    def then_period_was_rejected(self, period):
+        self.assertIsInstance(self.result, ValueError)
+        self.assertEqual('Invalid argument: {}'.format(period), self.result.message)
 
 
 class GetIntersectingPeriodsTest(BaseTestCase):
+    def setUp(self):
+        super(GetIntersectingPeriodsTest, self).setUp()
+        self.intersected_dates = NotImplemented
+        self.expected_results = NotImplemented
 
-    def test_date_arguments_and_date_range_with_default_post_days(self):
-        low = datetime(2014, 6, 15)
-        high = datetime(2014, 6, 16)
-        dates = list(date.get_intersecting_periods(low, high))
-        self.assertEquals(1, len(dates))
-        self.assertEquals(low, dates[0])
+    @parameterized.expand([
+        param(low=datetime(2014, 6, 15), high=datetime(2014, 6, 16))])
+    def test_date_arguments_and_date_range_with_default_post_days(self, low, high):
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.given_dates(list(date.get_intersecting_periods(datetime(2014, 6, 15),
```

Comment:
You don't need `given_dates` or `given_date_parsed` here. Let's take a look at `list(date.get_intersecting_periods(datetime(2014, 6, 15), datetime(2014, 6, 16)))`. First, you can clearly see that datetimes you've used there are actually `low` and `high` parameters (which are already in the environment of the test, thanks to `given_*` calls above). So now we can rewrite it into `list(date.get_intersecting_periods(self.period_low, self.period_high))`. Next thing to consider is `date.get_intersecting_periods` is actually the method we want to check (test action), so according to what we previously discussed it should go into the `when_*` method. Luckily we already have `when_intersecting_period_calculated` we can use. :smiley: